### PR TITLE
grimoire free fights redux

### PR DESCRIPTION
**Pull request by Rinn**
replaces #1497 @horrible-little-slime 

In the interest in not causing a billion merge conflicts again, we could probably commit this as is and continue replacing more tasks in additional PRs

### CODE REVIEW

**Comment by Rinn** [comment]
i haven't run this yet

File: packages/garbo/src/tasks/freeFight.ts [hunk 1]
```diff
@@ -0,0 +1,612 @@
...
+import { GarboTask } from "./engine";
+
+type GarboFreeFightTask = GarboTask & {
+  combatCount: () => number;
+  tentacle: boolean; // if a tentacle fight can follow
+};
+
+const DEFAULT_FREE_FIGHT_TASK = {
+  // GarboTask
+  combat: new CombatStrategy().autoattack(Macro.basicCombat()),
```
Comment (horrible-little-slime) [inline packages/garbo/src/tasks/freeFight.ts:64]: We have a `GarboStrategy` class that does a lot of the autoattack and macro handling for us; you can just do `new GarboStrategy(Macro.basicCombat())`

File: packages/garbo/src/tasks/freeFight.ts [hunk 2]
```diff
@@ -0,0 +1,612 @@
...
+  limit: { skip: 1 },
+  outfit: freeFightOutfit,
+  spendsTurn: false,
```
Comment (horrible-little-slime) [inline packages/garbo/src/tasks/freeFight.ts:67]: We can make some of these defaults on the engine instance rather than being here.

**Review by pstalcup — DISMISSED** [review]

**Comment by Rinn** [comment]
Okay I ran this and fixed a few minor things and all the tasks I converted so far work fine.

File: packages/garbo/src/tasks/freeFight.ts [hunk 3]
```diff
@@ -0,0 +1,614 @@
...
+    },
+  ],
+  [$skill`Soul Bubble`, () => (mySoulsauce() >= 5 ? 2 : 0)],
+  [$skill`Summon Love Gnats`, 1],
+  [$item`Rain-Doh blue balls`, 1],
+]);
+
+const FreeFightTasks: GarboFreeFightTask[] = [
+  {
+    ...DEFAULT_FREE_FIGHT_TASK,
```
Comment (pstalcup) [inline packages/garbo/src/tasks/freeFight.ts:133]: I think there are some clever typescript ways of doing this wihtout having hte same ... in every entry

**Review by pstalcup — APPROVED** [review]

**Pull request merged**